hook-worker/src/worker.rs

```diff
@@ -165,65 +172,87 @@ impl<'p> WebhookWorker<'p> {
         if transactional {
             loop {
                 report_semaphore_utilization();
-                let webhook_job = self.wait_for_job_tx().await;
-                spawn_webhook_job_processing_task(
-                    self.client.clone(),
-                    semaphore.clone(),
-                    self.retry_policy.clone(),
-                    webhook_job,
-                )
-                .await;
+                let mut batch = self.wait_for_jobs_tx().await;
+
```

Comment:
nit, I'd love to have a histogram of the real batch size, to check whether we're hitting it and could benefit from raising it.